mathics/builtin/quantum_mechanics/angular.py

```diff
@@ -0,0 +1,220 @@
+"""
+Angular Momentum
+
+<url>:Angular momentum: https://en.wikipedia.org/wiki/Angular_momentum</url> in physics is the rotational analog of linear momentum. It is an important quantity in physics because it is a conserved quantity the total angular momentum of a closed system remains constant.
+"""
+
+from typing import List
+
+from sympy.physics.matrices import msigma
+from sympy.physics.quantum.cg import CG
+from sympy.physics.wigner import wigner_3j, wigner_6j
+
+from mathics.builtin.base import SympyFunction
+from mathics.core.atoms import Integer
+from mathics.core.convert.sympy import from_sympy
+from mathics.core.evaluation import Evaluation
+from mathics.core.list import ListExpression
+from mathics.core.symbols import Symbol
+
+from mathics.core.attributes import (
+    listable as A_LISTABLE,
+    # numeric_function as A_NUMERIC_FUNCTION,
+    protected as A_PROTECTED,
+    read_protected as A_READ_PROTECTED,
+)
+
+
+class ClebschGordan(SympyFunction):
+    """
+    <url>:Clebsch-Gordan coefficients matrices: https://en.wikipedia.org/wiki/Clebsch%E2%80%93Gordan_coefficients</url> (<url>:SymPy: https://docs.sympy.org/latest/modules/physics/quantum/cg.html</url>, <url>:WMA: https://reference.wolfram.com/language/ref/ClebschGordan</url>)
+
+    <dl>
+      <dt>'ClebschGordan[{$j1$, $m1$}, {$j2$, $m2$}, {$j$ $m$}]'
+      <dd>returns the Clebsch-Gordan coefficient for the decomposition of in terms of |$j$, $m$|$j2$, $m2$>.
+    </dl>
+
+    >> ClebschGordan[{3 / 2, 3 / 2}, {1 / 2, -1 / 2}, {1, 1}]
+     = Sqrt[3] / 2
+
+    Compare with WMA example:
+    >> ClebschGordan[{5, 0}, {4, 0}, {1, 0}] == Sqrt[5 / 33]
+     = True
+
+    """
+
+    attributes = A_PROTECTED | A_READ_PROTECTED
+    summary_text = "Clebsch-Gordan coefficient"
+    sympy_name = "physics.quantum.cg.CG"
+
+    def apply(
+        self,
+        j1m1: ListExpression,
+        j2m2: ListExpression,
+        jm: ListExpression,
+        evaluation: Evaluation,
+    ):
+        "ClebschGordan[j1m1_List, j2m2_List, jm_List]"
+
+        sympy_jms: List[int] = []
+
+        for pair in (j1m1, j2m2, jm):
+            if len(pair.elements) != 2:
+                return
+            sympy_jms += [p.to_sympy() for p in pair.elements]
+        return from_sympy(CG(*sympy_jms).doit())
+
+
```

Comment:
At some place, it would be good to implement also `KroneckerProduct` (https://reference.wolfram.com/language/ref/KroneckerProduct.html)

Reply:
I have been going by reference guide section. That seems to belong to Matrix or Tensor things even though everything is related. Feel free to add it, but I am not sure where we would add it.

Reply:
> I am not sure where we would add it.

It looks like it goes under "Vector Space Operations" which we already have. See https://reference.wolfram.com/language/guide/OperationsOnVectors.html

Reply:
As I was doing these, it occurred to me that there was a set not intuitive process I used that makes doing this kind of thing more automatic. 

* Find a Builtin that hasn't been implemented. This PR comes from the bottom this [this issue](https://github.com/mathics/Mathics/issues/1483), TODO's at the bottom of newer sections/modules list missing functions. 
* Decide where in the code/doc to put it. As noted,  I first noticed it as a missing Combinatorial Function. However on more recent reflection, those last three are more specifically isolated as Angular Momentum functions (and in Combinatorial Functions it is not split out or noted as such). 
* See if it is a SymPy function. And the rest will assume yes. But if not try to scipi and mpmath. I suppose numpy might be tried too.  
* Copy one of the newer existing Sympy functions and adapt. Newer existing Sympy functions have Wiki, SyPy, and WMA links. Probably in the WMA link, all you will have to do is change the copied WMA name into the new name. 
* Use WMA link to adapt the docstring. It might be consulted to get the summary text too. 
* Go into mathics and see that you can get "info" or "?" on the changed doc. 
* Adapt the pattern match rule and parameters. This is probably the hardest part. `Cases` in mathics can help here.
* Adapt the eval method parameters and body. 
* Make sure the function attributes are correct. 
* Adapt some doctests and test. Examples from WMA and SymPy are a good source, but try to make the examples meaningful
* Run the tests using `python mathics/docpipeline -s <builtin name>` 
* Go over error conditions and error messages
* Write unit tests for all of these edge cases and uninteresting cases to try

The above even though long is still kind of sketchy. So I will go through this process on KroneckerProdect as a detailed example and put this in the developer guide.

Reply:
I started down this path. See https://github.com/Mathics3/mathics-development-guide/tree/adding-KroneckerProduct and https://mathics-development-guide.readthedocs.io/en/adding-kroneckerproduct/extending/developing-code/extending/adding-new-mathics-function.html. 

I stopped however when I see that there is this is not going to make for a simple example, because of the matrix checking and conversion in SymPy that we probably need to do. 

I will pick a simpler example. As for that KroneckerProduct branch, I don't intend on going further there.

Reply:
In that case, I would say that this is almost ready.

Reply:
(I never was my intention to add it here - these are the angular momentum functions.)

Possibly over the weekend I may add pytests.. However redoing how Part works and changes ListExpression is more important.